3p/integration.js

```diff
@@ -375,21 +387,22 @@ const defaultAllowedTypesInCustomFrame = [
 
 
 /**
- * Gets data encoded in iframe name attribute.
- * @return {!JsonObject}
+ * Initialize 3p frame.
+ * @param {!Window} win
  */
-function getData(iframeName) {
-  try {
-    // TODO(bradfrizzell@): Change the data structure of the attributes
-    //    to make it less terrible.
-    return parseJson(iframeName)['attributes'];
-  } catch (err) {
-    if (!getMode().test) {
-      dev().info(
-          'INTEGRATION', 'Could not parse context from:', iframeName);
-    }
-    return FALLBACK_CONTEXT_DATA;
-  }
+function init(win) {
+  const data = getData();
```

Comment:
I disagree with `mode`. Experiment toggles don't apply here. "Context" is not really a part of "configuration". How about:

```
- context
   - data
   - (others)
- config
   - mode
   - experiment toggles
   - (others)
```

By splitting context and config we're creating a clear boundary between "AMP land" (config) and "frame land" (context). We would also have two clear functions that would be only be invoked once (`get3pAmpConfig`, `get3pContextData`, names TBD). 

We don't need a module-level global, I find that a little ugly. `once`'ing something like `parseFrameName` would accomplish the same thing without global state, which I find a little more readable and manageable.